File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/118-PascalsTriangle/118-PascalsTriangle.java
// Last updated: 8/4/2026, 7:21:52 PM
class Solution {
     public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        for (int i = 1; i <= numRows; i++) {
            result.add(generateRow(i));
        }
        return result;
    }

    List<Integer> generateRow(int row) {
        List<Integer> ansRow = new ArrayList<>();
        int ans = 1;
        ansRow.add((int) ans);

        for (int i = 1; i < row; i++) {
            ans = ans * (row - i);
            ans = ans / i;
            ansRow.add(ans);
        }
        return ansRow;
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/118-PascalsTriangle/118-PascalsTriangle.cpp
// Last updated: 8/4/2026, 7:21:59 PM
class Solution {
public:

   vector<int> generateRow(int row)
{
    long long ans = 1;
    vector<int> ansRow;
    ansRow.push_back(1);
    for(int i=1;i<row;i++)
    {
        ans = ans * (row - i);
        ans = ans / i;
        ansRow.push_back(ans);
    }
    return ansRow;
}
    vector<vector<int>> generate(int numRows) {
         vector<vector<int>> v;
        for(int i=1;i<=numRows;i++)
        {
            vector<int> temp = generateRow(i);
            v.push_back(temp);
        }
        return v;
    }
};